### PR TITLE
added serialization to exceptions

### PR DESCRIPTION
**Pull request by darylvL**

### CODE REVIEW

**Comment by NicolaasVB** [comment]
I don't see tests

File: src/Narato.ResponseMiddleware.Models/Exceptions/ValidationException.cs [hunk 1]
```diff
@@ -23,5 +25,16 @@ public ValidationException(string message, ModelValidationDictionary<T> validati
...
             ValidationMessages = validationMessages;
         }
+
+        //Deserialization constructor.
+        public ValidationException(SerializationInfo info, StreamingContext context)
+            : base(info, context) { }
+
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            base.GetObjectData(info, context);
```
Comment (NicolaasVB) [inline src/Narato.ResponseMiddleware.Models/Exceptions/ValidationException.cs]: pretty sure you need to add the validationMessages to the info here

File: src/Narato.ResponseMiddleware.Models/Exceptions/ValidationException.cs [hunk 2]
```diff
@@ -23,5 +25,16 @@ public ValidationException(string message, ModelValidationDictionary<T> validati
         {
             ValidationMessages = validationMessages;
         }
+
+        //Deserialization constructor.
+        public ValidationException(SerializationInfo info, StreamingContext context)
+            : base(info, context) { }
```
Comment (NicolaasVB) [inline src/Narato.ResponseMiddleware.Models/Exceptions/ValidationException.cs]: pretty sure you need to set the validationMessages here

File: src/Narato.ResponseMiddleware.Models/Exceptions/EntityNotFoundException.cs [hunk 3]
```diff
@@ -1,14 +1,25 @@
 ﻿using System;
+using System.Runtime.Serialization;
 
 namespace Narato.ResponseMiddleware.Models.Exceptions
 {
-    public class EntityNotFoundException : CodedException
+    [Serializable]
+    public class EntityNotFoundException : CodedException, ISerializable
```
Comment (NicolaasVB) [inline src/Narato.ResponseMiddleware.Models/Exceptions/EntityNotFoundException.cs]: no need to implement ISerializable, CodedException already implements it

Comment (NicolaasVB) [reply src/Narato.ResponseMiddleware.Models/Exceptions/EntityNotFoundException.cs]: did you test this?... I can't accept this PR if you ignore comments

File: test/Narato.ResponseMiddleware.Models.Test/Serialization/ExceptionSerializationTest.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        public void EntityNotFoundSerializationTest()
+        {
+            var ErrorCode = "Test";
+            var exception = new EntityNotFoundException(ErrorCode, "Test Message");
+            try {
+                var deserializedException = Clone(exception);
+                Assert.Equal(deserializedException.ErrorCode, ErrorCode);
+            }
+            catch(Exception ex) {
+                Assert.Null(ex.Message);
```
Comment (NicolaasVB) [inline test/Narato.ResponseMiddleware.Models.Test/Serialization/ExceptionSerializationTest.cs]: This is a very weird way of testing... Don't do a try-catch in a test method! Either you expect an exception, and you use Assert.Throws<>()
or you don't expect an exception, and you just do what you do within the try clause now

this comment applies to every test method you wrote :)

**Review by NicolaasVB — COMMENTED** [review]
check my comments

File: test/Narato.ResponseMiddleware.Models.Test/Serialization/ExceptionSerializationTest.cs (hunk 4 above)
Comment (darylvL) [reply test/Narato.ResponseMiddleware.Models.Test/Serialization/ExceptionSerializationTest.cs]: Good point :)

**Pull request merged**